taichi/ir/transforms.h

```diff
@@ -114,6 +114,11 @@ void optimize_bit_struct_stores(IRNode *root,
                                 const CompileConfig &config,
                                 AnalysisManager *amgr);
 
+enum class ExternalPtrAccess : int { NONE = 0, READ = 1, WRITE = 2 };
+
+std::unordered_map<int, int> detect_external_ptr_access_in_task(
```

Comment:
Since this counts as an API, let's be explicit about the return types. Also would you add a docstring to briefly explain this (something like https://github.com/taichi-dev/taichi/blob/c2bdc82f8f62b74be93ba7df8e9c871b389fc477/taichi/ir/analysis.h#L66-L78) ? 

```suggestion
std::unordered_map<int, ExternqalPtrAccess> detect_external_ptr_access_in_task(
```